pkg/cmd/codespace/ssh.go

```diff
@@ -147,6 +177,129 @@ func (a *App) SSH(ctx context.Context, sshArgs []string, opts sshOptions) (err e
 	}
 }
 
+func (a *App) ListOpensshConfig(ctx context.Context, opts sshOptions) error {
+	// Ensure all child tasks (e.g. port forwarding) terminate before return.
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var err error
+	var codespaces []*api.Codespace
+	if opts.codespace == "" {
+		a.StartProgressIndicatorWithLabel("Fetching codespaces")
+		codespaces, err = a.apiClient.ListCodespaces(ctx, -1)
+		a.StopProgressIndicator()
+	} else {
+		var codespace *api.Codespace
+		codespace, err = getOrChooseCodespace(ctx, a.apiClient, opts.codespace)
+		codespaces = []*api.Codespace{codespace}
+	}
+	if err != nil {
+		return fmt.Errorf("error getting codespace info: %w", err)
+	}
+
+	t, err := template.New("ssh_config").Parse(`Host cs.{{.Name}}.{{.EscapedRef}}
+	User {{.SshUser}}
+	ProxyCommand {{.GHExec}} cs ssh -c {{.Name}} --stdio
+	UserKnownHostsFile=/dev/null
+	StrictHostKeyChecking no
+	LogLevel quiet
+	ControlMaster auto
+
+`)
+	if err != nil {
+		return fmt.Errorf("error formatting template: %w", err)
+	}
+
+	ghexec, err := os.Executable()
+	if err != nil {
+		return err
+	}
+
+	sshUsers := map[string]string{}
+
+	for _, cs := range codespaces {
```

Comment:
It feels odd to me that `gh cs ssh` would perform any operation on the complete list of your codespaces, since in its normal mode of operation it only ever handled a single codespace at a time. I wonder whether the `--config` functionality warrants its own dedicated command, since it completely changes the semantics of the `ssh` command?

Reply:
I have it coded so that `--config` honors `-c`/`--codespace`, giving the option to either generate a full config for all codespaces, or the config for a single codespace. The details of how someone would use these depends on how folks integrate this into their workflow.

Ideally, nobody would have to invoke `gh cs ssh --config` manually at all - if a config were instead kept up-to-date in a well-known location (whenever `gh cs` or vscode get information about codespace creation/destruction), then users could simply include that file into their ssh config and have it be always up to date.

I agree with you about this perhaps warranting a separate command.

Reply:
moved to a subcommand: https://github.com/cli/cli/pull/4915/commits/937da7175586d008ecc02296e66b011b8bc19a83

Reply:
I agree that this doesn't seem like a natural fit for the ssh command.

This loop should be done in parallel. (You'll want to run the progress indicator around the whole loop.)